lib/theme/text_theme.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'colors.dart';
+
+TextTheme textTheme() {
+  return const TextTheme(
+    // body text 1
+    bodyLarge: TextStyle(
+      color: AllAppColors.labelLight,
+      fontSize: 16,
+      letterSpacing: 0.2,
+    ),
+
+    // body text 2
+    bodyMedium: TextStyle(
+      color: AllAppColors.labelLight,
+      fontSize: 14,
+      letterSpacing: 0.22
+    ),
+
+    // button text 2
+    labelLarge: TextStyle(
+      color: AllAppColors.labelLight,
+      fontWeight: FontWeight.bold,
+      fontSize: 14,
+      letterSpacing: 0.32
+    ),
+
+    // button text 3
```

Comment:
These comments are not of any value. They don't add anything that the code doesn't already clarify?

Reply:
These comments match with the namings of the typography in the figma designs we are following. I found that the named typography used in figma design is bit different than the one available with TextTheme widget. To avoid any confusion I added these comments.

Reply:
Okay that's fair enough. What Can you give me examples of what the Figma color is called and what it's called here?Because if we follow material guidelines the names should be the same everywhere.

Reply:
Currently flutter's TextTheme class is based on material 3 design typography. Hence there are many deprecated properties of TextTheme class (like headline <1-6>) Large, Medium, Small convention is used in design typography of Material 3 in contrast to numbers in design typography Material 2.

But, are you sure that the typography used in Figma design is following Material Design Typography? Because there are many TextStyles (like Small, Body <3-4>, Caption 2, Overline 2, subHeadline) that don't match to that defined in material design typography. Here are my supporting resources (Material 2):
https://m2.material.io/design/typography/the-type-system.html#type-scale
https://api.flutter.dev/flutter/material/TextTheme-class.html

Currently, I'm leading to a point, where I'm confused if the typography is Figma design is following Material Design Spec or not. If it is still following, then what point am I missing about understanding material design spec? Can the nomenclature be changed as per our needs, and can the number of styles increase/decreast, but the typography can still belong to Material Design (2, here)?

Reply:
Added documentation for this.